<?xml version="1.0"?>
<launch>
    <include file="$(find jetracer)/launch/jetracer.launch" />
    <include file="$(find jetracer)/launch/lidar.launch" />

    <node pkg="tf2_ros" type="static_transform_publisher" name="base_footprint_to_base_link" 
          args="0 0 0 0 0 0 base_footprint base_link" />

     <node name="local_costmap" pkg="costmap_2d" type="costmap_2d_node" output="screen">
		<rosparam file="$(find combined_system)/config/local_costmap_params.yaml" command="load" ns="costmap"/>
	</node>

    <node pkg="gmapping" type="slam_gmapping" name="slam_mapping" output="screen">
        <param name="map_frame" value="map"/>
        <param name="odom_frame" value="odom"/>
        <param name="base_frame" value="base_link"/>
        <param name="map_update_interval" value="1.0"/>
        <remap from="scan" to="/scan"/>
    </node>

    <node pkg="combined_system" type="state_machine" name="state_machine" output="screen">
		<rosparam command="load" file="$(find combined_system)/config/planner_params.yaml" />
    </node>
</launch>